post_office/migrations/0010_message_id.py

```diff
@@ -0,0 +1,38 @@
+import random
+from django.db import migrations, models
+
+from post_office.models import STATUS
+from post_office.settings import get_message_id_on_delivery, get_message_id_fqdn
+
+def forwards(apps, schema_editor):
+    if get_message_id_on_delivery():
+        return
+    msg_id_fqdn = get_message_id_fqdn()
+    Email = apps.get_model('post_office', 'Email')
+    for email in Email.objects.using(schema_editor.connection.alias).filter(message_id__isnull=True):
+        if email.status in [STATUS.queued, STATUS.requeued]:
+            # create a unique Message-ID for all emails which have not been send yet
+            randint1, randint2 = random.getrandbits(64), random.getrandbits(16)
+            email.message_id = '<{}.{}.{}@{}>'.format(email.id, randint1, randint2, msg_id_fqdn)
+            email.save()
```

Comment:
I'm not sure we should do a forwards migration in this case. This will cause huge spikes in DB writes when migrating DBs with a large number of emails. I think we should just drop this and leave the migration part optional for people who need it.

Reply:
well, it is optional. If you don't set the configuration directive `MESSAGE_ID_ON_DELIVERY` to `False`, then nothing happens.